main.go

```diff
@@ -52,6 +52,8 @@ type Class struct {
 type Product struct {
 	// vendor ID for the product
 	VendorID string `json:"vendor_id"`
+	// vendor name for the product
+	VendorName string `json:"vendor_name"`
```

Comment:
I've designed the main structs in `pcidb` to be a normalized form of data. In other words, there are no fields in a struct that are attributes of a related struct. In this case, `VendorName` is an attribute of the `Vendor` struct which can be "looked up" via the `Product.VendorID` field. I do this to minimize the amount of memory consumed by the `pcidb.PCIDB` struct. In this case, `Product.VendorName` would be duplicated information to `Vendor.Name`.

You can use the `pcidb.PCIDB` struct to get the vendor's name from your `pcidb.Product` by doing this:

```go
package main

import (
	"fmt"

	"github.com/jaypipes/pcidb"
)

func main() {
	db, err := pcidb.New()
	if err != nil {
		panic(err)
	}
	x := 0
	for productID, product := range db.Products {
		if x > 4 {
			break
		}
		fmt.Println(
			"product ID:", productID,
			"vendor name:", db.Vendors[product.VendorID].Name,
		)
		x++
	}
}
```

```
jaypipes@lappie:~/pcidbexample$ go run tmp.go 
product ID: 126c1211 vendor name: Northern Telecom
product ID: 1415c5ad vendor name: Oxford Semiconductor Ltd
product ID: 14e44705 vendor name: Broadcom Inc. and subsidiaries
product ID: 808604d0 vendor name: Intel Corporation
product ID: 10222095 vendor name: Advanced Micro Devices, Inc. [AMD]
```

So, I'm afraid I'm going to have to say no to this particular contribution @vultj. I hope you understand.